unit-tests/rsutils/test-version.py

```diff
@@ -0,0 +1,229 @@
+    test.check_false( version( "1.2.3.4" ) > version( "1.2.3.5" ))
+    test.check_false( version( "1.2.3.4" ) > version( "1.2.3.4" ))
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Operator <" )
+try:
+    test.check( version( "1.2.3.4" ) < version( "2.2.3.4" ))
+    test.check( version( "1.2.3.4" ) < version( "1.3.3.4" ))
+    test.check( version( "1.2.3.4" ) < version( "1.2.4.4" ))
+    test.check( version( "1.2.3.4" ) < version( "1.2.3.5" ))
+
+    test.check_false( version( "1.2.3.4" ) < version( "0.2.3.4" ))
+    test.check_false( version( "1.2.3.4" ) < version( "1.1.3.4" ))
+    test.check_false( version( "1.2.3.4" ) < version( "1.2.2.4" ))
+    test.check_false( version( "1.2.3.4" ) < version( "1.2.3.3" ))
+
+    test.check_false( version( "1.2.3.4" ) < version( "1.2.3.4" ))
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Operator >=" )
+try:
+    test.check( version( "1.2.3.4" ) >= version( "1.2.3.4" ))
+
+    test.check( version( "2.2.3.4" ) >= version( "1.2.3.4" ))
+    test.check( version( "1.3.3.4" ) >= version( "1.2.3.4" ))
+    test.check( version( "1.2.4.4" ) >= version( "1.2.3.4" ))
+    test.check( version( "1.2.3.5" ) >= version( "1.2.3.4" ))
+
+    test.check_false( version( "1.2.3.4" ) >= version( "2.2.3.4" ))
+    test.check_false( version( "1.2.3.4" ) >= version( "1.3.3.4" ))
+    test.check_false( version( "1.2.3.4" ) >= version( "1.2.4.4" ))
+    test.check_false( version( "1.2.3.4" ) >= version( "1.2.3.5" ))
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Operator <=" )
+try:
+    test.check( version( "1.2.3.4" ) <= version( "1.2.3.4" ))
+
+    test.check( version( "1.2.3.4" ) <= version( "2.2.3.4" ))
+    test.check( version( "1.2.3.4" ) <= version( "1.3.3.4" ))
+    test.check( version( "1.2.3.4" ) <= version( "1.2.4.4" ))
+    test.check( version( "1.2.3.4" ) <= version( "1.2.3.5" ))
+
+    test.check_false( version( "2.2.3.4" ) <= version( "1.2.3.4" ))
+    test.check_false( version( "1.3.3.4" ) <= version( "1.2.3.4" ))
+    test.check_false( version( "1.2.4.4" ) <= version( "1.2.3.4" ))
+    test.check_false( version( "1.2.3.5" ) <= version( "1.2.3.4" ))
+except:
+    test.unexpected_exception()
+test.finish()
+#
+#############################################################################################
+#
+test.start( "Is between" )
+try:
+    v1 = version( "01.02.03.04" )
+    v2 = version( "01.03.03.04" )
+    v3 = version( "02.01.03.04" )
+
+    test.check( v2.is_between( v1, v3 ))
+    test.check( v3.is_between( v1, v3 ))
+    test.check( v2.is_between( v2, v3 ))
+    test.check( v2.is_between( v2, v2 ))
```

Comment:
Better to add a state when `is_between()` failed

Reply:
Hey, I just took your code from the .cpp :)